docs/user-guide/deployments.md

```diff
@@ -468,7 +468,15 @@ To learn more about when a pod is considered ready, see [Container Probes](/docs
 
 ### Revision History Limit
 
-`.spec.revisionHistoryLimit` is an optional field that specifies the number of old replica sets to retain to allow rollback. All old replica sets will be kept by default, if this field is not set. The configuration of each Deployment revision is stored in its replica sets; therefore, once an old replica set is deleted, you lose the ability to rollback to that revision of Deployment. 
+`.spec.revisionHistoryLimit` is an optional field that specifies the number of
+old replica sets to retain to allow rollback. Its ideal value depends on the
+frequency and stability of new deployments. All old replica sets will be kept by
+default, consuming resources in `etcd` and crowding the output of `kubectl get
+rs`, if this field is not set. The configuration of each Deployment revision is
+stored in its replica sets; therefore, once an old replica set is deleted, you
+lose the ability to roll back to that revision of Deployment. More specifically,
+setting this field to zero means that a new deployment revision can be undone
```

Comment:
We can say that the deployment revision history is stored in the RS it controls in the beginning of this paragraph, since this field is "revision history limit" and it makes sense to talk about revision first, RSes later.